pkg/specgen/generate/container_create.go

```diff
@@ -21,6 +21,27 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+func platformToOCIVariant(arch string, os string, defaultRuntime libpod.OCIRuntime) string {
+	variant := defaultRuntime.Name()
+	if arch == "wasm32" || arch == "wasm64" || os == "wasi" {
+		variant = "crun-wasm"
+	}
+	return variant
+}
+
+func modifyCrunIfNeeded(conf libpod.OCIRuntime, variant string) (string, string) {
+	var nameReplace, pathReplace string
+	if variant != "crun" {
```

Comment:
This still seems rather complex, Why not make the call above return "", or return the path and then you are done.

Reply:
Simplified.